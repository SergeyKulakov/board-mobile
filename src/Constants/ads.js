import { Platform } from 'react-native'
import {
  FB_ADS_PLACEMENT_ID_IOS,
  FB_ADS_PLACEMENT_ID_ANDROID,
  ADMOB_UNIT_ID_IOS,
  ADMOB_UNIT_ID_ANDROID,
} from 'react-native-dotenv'

export const admobUnitId = Platform.select({
  // ios: 'ca-app-pub-3940256099942544/1033173712', // test unit id
  ios: ADMOB_UNIT_ID_IOS,
  android: ADMOB_UNIT_ID_ANDROID,
})

export const fbAdsPlacementId = Platform.select({
  ios: FB_ADS_PLACEMENT_ID_IOS,
  android: FB_ADS_PLACEMENT_ID_ANDROID,
})
