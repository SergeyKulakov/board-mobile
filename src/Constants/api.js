import {
  API_URL,
  SECOND_API,
  CLIENT_URL,
  APP_BUNDLE_ID,
  LOGO_PATH,
} from 'react-native-dotenv'

export const apiUrl = API_URL
export const secondApi = SECOND_API
export const clientUrl = CLIENT_URL
export const appBundleId = APP_BUNDLE_ID
export const logoPath = LOGO_PATH
