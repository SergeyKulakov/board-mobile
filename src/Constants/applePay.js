import { MERCHANT_IDENTIFIER } from 'react-native-dotenv'

const METHOD_DATA = [
  {
    supportedMethods: ['apple-pay'],
    data: {
      merchantIdentifier: MERCHANT_IDENTIFIER,
      supportedNetworks: ['visa', 'mastercard', 'amex'],
      countryCode: 'US',
      currencyCode: 'USD',
    },
  },
]

const OPTIONS = {}

export default {
  METHOD_DATA,
  OPTIONS,
}
