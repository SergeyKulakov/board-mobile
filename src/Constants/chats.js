const blockChatConst =
  '+NjSRRW1lPggeqleTXDt/HtjO4MqkwL3pNcPzkPcL7No1ih5Jm6G6dp1uzZd233JZRnjpUt4fgBwH'
const unblockChatConst =
  'pgciFniskM6vHj7TPg5eMoDe2MRe58jBBkMHKlLou9qMPZxENADgegHNpHBKAPMZbaFKL0Wbp6'

export { blockChatConst, unblockChatConst }
