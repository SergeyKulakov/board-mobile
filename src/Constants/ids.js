export const API_CALL = 'API_CALL'
export const SOCKET_EVENT = 'SOCKET_EVENT'
export const API_CALL_FAILURE = 'API_CALL_FAILURE'
