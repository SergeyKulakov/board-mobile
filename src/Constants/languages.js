export const languages = [
  {
    type: 'en',
    flag: 'english',
    name: 'English',
  },
  {
    type: 'fr',
    flag: 'french',
    name: 'French',
  },
  {
    type: 'es',
    flag: 'spanish',
    name: 'Spanish',
  },
  {
    type: 'it',
    flag: 'italian',
    name: 'Italian',
  },
  {
    type: 'pt',
    flag: 'portuguese',
    name: 'Portuguese',
  },
  {
    type: 'de',
    flag: 'deutsch',
    name: 'German',
  },
  {
    type: 'ru',
    flag: 'russian',
    name: 'Russian',
  },
  {
    type: 'hi',
    flag: 'indian',
    name: 'Hindi',
  },
  {
    type: 'bg',
    flag: 'indian',
    name: 'Bengali',
  },
  {
    type: 'te',
    flag: 'indian',
    name: 'Telugu',
  },
  {
    type: 'mr',
    flag: 'indian',
    name: 'Marathi',
  },
  {
    type: 'ta',
    flag: 'indian',
    name: 'Tamil',
  },
  {
    type: 'gu',
    flag: 'indian',
    name: 'Gujarati',
  },
  {
    type: 'kn',
    flag: 'indian',
    name: 'Kannada',
  },
  {
    type: 'ml',
    flag: 'indian',
    name: 'Malayalam',
  },
]
