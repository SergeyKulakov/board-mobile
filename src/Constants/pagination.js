export const JOBS_ELEMENTS_PER_PAGE = 20
export const SERVICE_PROVIDERS_ELEMENTS_PER_PAGE = 20
