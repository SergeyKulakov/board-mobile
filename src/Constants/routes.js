export const login = 'login'
export const initial = 'initial'
export const singUp = 'singUp'
export const forgotPass = 'forgotPass'
export const home = 'home'
export const profile = 'profile'
export const locationModal = 'modal.location'
export const servicesModal = 'modal.services'
export const sideMenu = 'sideMenu'
export const confirmSignUp = 'confirmSignUp'
export const rootLoading = 'rootLoading'
export const confirmResetPassword = 'confirmResetPassword'
export const notifications = 'notifications'
export const findJobs = 'findJobs'
export const filterJobs = 'filterJobs'
export const filterSubServices = 'modal.filterSubServices'
export const googleMapModal = 'model.googleMap'
export const verificationsPhoneOrEmail = 'modal.verificationsPhoneOrEmail'
export const jobDescription = 'jobDescription'
export const imagesSlider = 'imagesSlider'
export const postJob = 'postJob'
export const findHelp = 'findHelp'
export const currencyModal = 'modal.currency'
export const sortedModal = 'modal.sorted'
export const phonesModal = 'modal.phones'
export const serviceProviderProfile = 'serviceProviderProfile'
export const searchModal = 'model.searchModal'
export const myFavourites = 'myFavourites'
export const myJobs = 'myJobs'
export const canceledModal = 'canceledModal'
export const seePeopleWhoApplied = 'seePeopleWhoApplied'
export const selectJob = 'selectJob'
export const myRequests = 'myRequests'
export const ratingReview = 'ratingReview'
export const trackJobMap = 'trackJobMap'
export const track = 'track'
export const chats = 'chats'
export const chat = 'chat'
export const settings = 'settings'
export const deleteMyAccount = 'deleteMyAccount'
export const mapModal = 'mapModal'
export const subscriptions = 'subscriptions'
export const reviews = 'reviews'
export const report = 'report'
export const onBoarding = 'onBoarding'
