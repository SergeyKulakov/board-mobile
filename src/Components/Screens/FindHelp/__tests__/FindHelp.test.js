import React from 'react'
import { shallow } from 'enzyme'
import * as routes from 'Constants/routes'

import 'jest-styled-components'

import Component from '../FindHelp'

describe('FindHelp', () => {
  const defaultProps = {
    componentId: 'findHelp',
    navigate: {
      showMessage: () => null,
      push: () => null,
      showModal: () => null,
      pop: () => null,
      showSidebar: () => null,
    },
    isEndList: false,
    filters: {},
    page: 0,
    user: {
      defaultRadius: '10',
    },
    serviceProviders: [],
    onLoadList: () => null,
    onAddFavorite: () => null,
    onDeleteFavorite: () => null,
    onRemoveServiceProvider: () => null,
    onSendRequest: () => null,
  }
  const serverError = {
    payload: {
      code: 'code',
    },
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    const withMapView = shallow(<Component {...defaultProps} />)
    withMapView.setState({ tabIndex: 1 })

    const renderMapServiceProviderCard = wrapper
      .instance()
      .renderMapServiceProviderCard({
        item: {
          username: 'username',
          picsOfWork: [],
          _id: 'username',
        },
      })

    expect([
      wrapper,
      withMapView,
      renderMapServiceProviderCard,
    ]).toMatchSnapshot()
  })

  it('reload data', () => {
    const props = {
      ...defaultProps,
      onLoadList: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.onLoadList).toHaveBeenCalledTimes(1)

    wrapper.setState({
      loading: {
        favouriteId: false,
        serviceProviders: false,
      },
    })

    wrapper
      .find('ServiceProvidersList')
      .props()
      .onRefresh()

    expect(props.onLoadList).toHaveBeenCalledTimes(2)
  })

  it('should show profile reviews', () => {
    const props = {
      ...defaultProps,
      navigate: {
        ...defaultProps.navigate,
        push: jest.fn(),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.navigate.push).not.toHaveBeenCalled()

    const profile = {
      username: 'userId',
    }

    wrapper.instance().handleOpenReviews(profile)

    expect(props.navigate.push).toHaveBeenCalledTimes(1)
    expect(props.navigate.push).toHaveBeenCalledWith(routes.reviews, {
      userId: 'userId',
    })
  })

  it('open sp profile', () => {
    const props = {
      ...defaultProps,
      navigate: {
        ...defaultProps.navigate,
        push: jest.fn(),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.navigate.push).not.toHaveBeenCalled()

    const profile = {
      username: 'userId',
    }

    wrapper.instance().handleOpenServiceProvider(profile)

    expect(props.navigate.push).toHaveBeenCalledTimes(1)
    expect(props.navigate.push).toHaveBeenCalledWith(
      routes.serviceProviderProfile,
      {
        userId: 'userId',
      },
    )
  })

  describe('search modal actions', () => {
    it('submit with keywords', done => {
      const props = {
        ...defaultProps,
        navigate: {
          ...defaultProps.navigate,
          showModal: jest.fn((route, { onSubmit }) => {
            setTimeout(() => {
              onSubmit({ keywords: 'keywords' })
            }, 1000)
          }),
        },
        onLoadList: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({
        loading: {
          favouriteId: false,
          serviceProviders: false,
        },
      })

      wrapper.instance().handleOpenSearchModal()

      expect(props.navigate.showModal).toHaveBeenCalledTimes(1)

      setTimeout(() => {
        expect(props.onLoadList).toHaveBeenCalledTimes(2)
        done()
      }, 1100)
    })

    it('submit with service', done => {
      const props = {
        ...defaultProps,
        navigate: {
          ...defaultProps.navigate,
          showModal: jest.fn((route, { onSubmit }) => {
            setTimeout(() => {
              onSubmit({
                service: {
                  _id: '12',
                  title: 'title',
                },
              })
            }, 1000)
          }),
        },
        onLoadList: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({
        loading: {
          favouriteId: false,
          serviceProviders: false,
        },
      })

      wrapper.instance().handleOpenSearchModal()

      expect(props.navigate.showModal).toHaveBeenCalledTimes(1)

      setTimeout(() => {
        expect(props.onLoadList).toHaveBeenCalledTimes(2)
        done()
      }, 1100)
    })
  })

  describe('remove profile card', () => {
    it('from list', () => {
      const props = {
        ...defaultProps,
        onRemoveServiceProvider: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      expect(props.onRemoveServiceProvider).not.toHaveBeenCalled()

      wrapper.setState({ tabIndex: 0 })

      const profile = {
        username: 'userId',
      }

      wrapper.instance().handleRemoveServiceProvider(profile)

      expect(props.onRemoveServiceProvider).toHaveBeenCalledTimes(1)
      expect(props.onRemoveServiceProvider).toHaveBeenCalledWith('userId')
    })

    it('from map', () => {
      const mapRef = {
        current: {
          hideSlider: jest.fn(),
        },
      }
      const props = {
        ...defaultProps,
        onRemoveServiceProvider: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().map = mapRef

      expect(props.onRemoveServiceProvider).not.toHaveBeenCalled()

      wrapper.setState({ tabIndex: 1 })

      const profile = {
        username: 'userId',
      }

      wrapper.instance().handleRemoveServiceProvider(profile)

      expect(props.onRemoveServiceProvider).not.toHaveBeenCalled()
      expect(mapRef.current.hideSlider).toHaveBeenCalledTimes(1)
    })
  })

  it('sort modal submit', done => {
    const props = {
      ...defaultProps,
      navigate: {
        showModal: jest.fn((route, { onSubmit }) => {
          setTimeout(() => {
            onSubmit()
          }, 1000)
        }),
      },
      onLoadList: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().handleSortClick()

    wrapper.setState({
      loading: {
        favouriteId: false,
        serviceProviders: false,
      },
    })

    expect(props.navigate.showModal).toHaveBeenCalledTimes(1)
    expect(props.onLoadList).toHaveBeenCalledTimes(1)

    setTimeout(() => {
      expect(props.onLoadList).toHaveBeenCalledTimes(2)
      done()
    }, 1100)
  })

  it('pagination', () => {
    const props = {
      ...defaultProps,
      page: 0,
      onLoadList: jest.fn(),
      isEndList: false,
      filters: {},
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.setState({
      loading: {
        favouriteId: false,
        serviceProviders: false,
      },
    })

    wrapper.instance().handleLoadServiceProviders = jest.fn()

    wrapper.instance().handleLoadMore()

    expect(wrapper.instance().handleLoadServiceProviders).toHaveBeenCalledTimes(
      1,
    )
    expect(wrapper.instance().handleLoadServiceProviders).toHaveBeenCalledWith(
      {},
      1,
    )
  })

  it('open images modal', () => {
    const props = {
      ...defaultProps,
      navigate: {
        ...defaultProps.navigate,
        showModal: jest.fn(),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    const args = [['sdasd', '13132r qef', '2131sda'], 1]

    wrapper.instance().handleOpenImages(...args)

    expect(props.navigate.showModal).toHaveBeenCalledTimes(1)
    expect(props.navigate.showModal).toHaveBeenCalledWith(routes.imagesSlider, {
      data: args[0],
      activeIndex: 1,
    })
  })

  describe('call favourite actions', () => {
    const getProps = isSuccessful => ({
      ...defaultProps,
      onAddFavorite: jest.fn((userId, callback) => {
        setTimeout(() => {
          callback({ error: isSuccessful ? undefined : serverError })
        }, 1000)
      }),
      onDeleteFavorite: jest.fn((favouriteId, callback) => {
        setTimeout(() => {
          callback({ error: isSuccessful ? undefined : serverError })
        }, 1000)
      }),
      getError: jest.fn(),
      onShowPuck: jest.fn(),
    })

    describe('add', () => {
      it('successful', done => {
        const props = getProps(true)

        const wrapper = shallow(<Component {...props} />)

        expect(props.onAddFavorite).not.toHaveBeenCalled()
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        expect(props.onShowPuck).not.toHaveBeenCalled()

        expect(wrapper.state('loading').favouriteId).toBe(null)

        const provider = {
          username: 'username',
          _id: 'userId',
        }

        wrapper.instance().handleFavouriteClick(provider)

        expect(wrapper.state('loading').favouriteId).toBe(provider._id)
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.onAddFavorite).toHaveBeenCalledTimes(1)

        setTimeout(() => {
          expect(wrapper.state('loading').favouriteId).toBe(null)
          expect(props.getError).not.toHaveBeenCalled()
          expect(props.onShowPuck).not.toHaveBeenCalled()
          done()
        }, 1100)
      })

      it('failure', done => {
        const props = getProps(false)

        const wrapper = shallow(<Component {...props} />)

        expect(props.onAddFavorite).not.toHaveBeenCalled()
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        expect(props.onShowPuck).not.toHaveBeenCalled()

        expect(wrapper.state('loading').favouriteId).toBe(null)

        const provider = {
          username: 'username',
          _id: 'userId',
        }

        wrapper.instance().handleFavouriteClick(provider)

        expect(wrapper.state('loading').favouriteId).toBe(provider._id)
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.onAddFavorite).toHaveBeenCalledTimes(1)

        setTimeout(() => {
          expect(wrapper.state('loading').favouriteId).toBe(null)
          expect(props.getError).toHaveBeenCalledTimes(1)
          expect(props.getError).toHaveBeenCalledWith(serverError)
          expect(props.onShowPuck).toHaveBeenCalledTimes(1)
          done()
        }, 1100)
      })
    })

    describe('remove', () => {
      it('successful', done => {
        const props = getProps(true)

        const wrapper = shallow(<Component {...props} />)

        expect(props.onAddFavorite).not.toHaveBeenCalled()
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        expect(props.onShowPuck).not.toHaveBeenCalled()

        expect(wrapper.state('loading').favouriteId).toBe(null)

        const provider = {
          username: 'username',
          _id: 'userId',
          favouriteId: '12341ewqedsae',
        }

        wrapper.instance().handleFavouriteClick(provider)

        expect(wrapper.state('loading').favouriteId).toBe(provider._id)
        expect(props.onDeleteFavorite).toHaveBeenCalledTimes(1)
        expect(props.onAddFavorite).not.toHaveBeenCalled()

        setTimeout(() => {
          expect(wrapper.state('loading').favouriteId).toBe(null)
          expect(props.getError).not.toHaveBeenCalled()
          expect(props.onShowPuck).not.toHaveBeenCalled()
          done()
        }, 1100)
      })

      it('failure', done => {
        const props = getProps(false)

        const wrapper = shallow(<Component {...props} />)

        expect(props.onAddFavorite).not.toHaveBeenCalled()
        expect(props.onDeleteFavorite).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        expect(props.onShowPuck).not.toHaveBeenCalled()

        expect(wrapper.state('loading').favouriteId).toBe(null)

        const provider = {
          username: 'username',
          _id: 'userId',
          favouriteId: '12341ewqedsae',
        }

        wrapper.instance().handleFavouriteClick(provider)

        expect(wrapper.state('loading').favouriteId).toBe(provider._id)
        expect(props.onDeleteFavorite).toHaveBeenCalledTimes(1)
        expect(props.onAddFavorite).not.toHaveBeenCalled()

        setTimeout(() => {
          expect(wrapper.state('loading').favouriteId).toBe(null)
          expect(props.getError).toHaveBeenCalledTimes(1)
          expect(props.getError).toHaveBeenCalledWith(serverError)
          expect(props.onShowPuck).toHaveBeenCalledTimes(1)
          done()
        }, 1100)
      })
    })
  })

  it('submit set filters', done => {
    const nextFilters = {
      address: 'address',
    }
    const props = {
      ...defaultProps,
      filters: {},
      navigate: {
        push: jest.fn((route, { onSubmit }) => {
          setTimeout(() => {
            onSubmit(nextFilters)
          }, 1000)
        }),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().handleLoadServiceProviders = jest.fn()

    wrapper.instance().handleOpenFilter()

    expect(props.navigate.push).toHaveBeenCalled()

    setTimeout(() => {
      expect(
        wrapper.instance().handleLoadServiceProviders,
      ).toHaveBeenCalledTimes(1)
      expect(
        wrapper.instance().handleLoadServiceProviders,
      ).toHaveBeenCalledWith(nextFilters, 0, true)
      done()
    }, 1100)
  })

  describe('load sp list', () => {
    const getProps = isSuccessful => ({
      ...defaultProps,
      onLoadList: jest.fn(({ callback }) => {
        setTimeout(() => {
          callback({ error: isSuccessful ? undefined : serverError })
        }, 1000)
      }),
      onShowPuck: jest.fn(),
      getError: jest.fn(),
    })

    it('successful', done => {
      const props = getProps(true)

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().handleLoadServiceProviders()

      expect(wrapper.state('loading')).toStrictEqual({
        serviceProviders: true,
        favouriteId: null,
        pagination: false,
      })

      expect(props.onLoadList).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('loading')).toStrictEqual({
          serviceProviders: false,
          favouriteId: null,
          pagination: false,
        })
        expect(props.onShowPuck).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        done()
      }, 1100)
    })

    it('failure', done => {
      const props = getProps(false)

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().handleLoadServiceProviders()

      expect(wrapper.state('loading')).toStrictEqual({
        serviceProviders: true,
        favouriteId: null,
        pagination: false,
      })

      expect(props.onLoadList).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('loading')).toStrictEqual({
          serviceProviders: false,
          favouriteId: null,
          pagination: false,
        })
        expect(props.onShowPuck).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledWith(serverError)
        done()
      }, 1100)
    })
  })
})
