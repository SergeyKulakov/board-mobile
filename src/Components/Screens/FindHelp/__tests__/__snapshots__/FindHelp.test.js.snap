// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindHelp render snapshot 1`] = `
Array [
  <Styled(View)>
    <Header
      activeScreenId={0}
      onClickBack={[Function]}
      onClickFilter={[Function]}
      onClickHamburger={[Function]}
      onClickNavItem={[Function]}
      onClickSearch={[Function]}
      onClickSort={[Function]}
      text={
        Object {
          "getTabTitle": [Function],
          "title": "Find Help",
        }
      }
    />
    <ServiceProvidersList
      data={Array []}
      favouriteLoadingId={null}
      isPaginationLoading={false}
      isRefreshing={true}
      onFavouriteClick={[Function]}
      onHireClick={[Function]}
      onImageClick={[Function]}
      onLoadMore={[Function]}
      onOpenReviews={[Function]}
      onProfileClick={[Function]}
      onRefresh={[Function]}
      onRemoveServiceProviderClick={[Function]}
      onShareClick={[Function]}
    />
    <ScreenLoader
      visible={false}
    />
  </Styled(View)>,
  <Styled(View)>
    <Header
      activeScreenId={1}
      onClickBack={[Function]}
      onClickFilter={[Function]}
      onClickHamburger={[Function]}
      onClickNavItem={[Function]}
      onClickSearch={[Function]}
      onClickSort={[Function]}
      text={
        Object {
          "getTabTitle": [Function],
          "title": "Find Help",
        }
      }
    />
    <Map
      data={Array []}
      extraData={
        Array [
          null,
        ]
      }
      isHelpCard={true}
      onLoadMore={[Function]}
      renderMapCard={[Function]}
      user={
        Object {
          "defaultRadius": "10",
        }
      }
    />
    <ScreenLoader
      visible={false}
    />
  </Styled(View)>,
  <ForwardRef>
    <ProviderProfileCard
      data={
        Object {
          "_id": "username",
          "picsOfWork": Array [],
          "username": "username",
        }
      }
      isFavouriteLoading={false}
      onClick={[Function]}
      onCommentClick={[Function]}
      onFavouriteClick={[Function]}
      onHireClick={[Function]}
      onImageClick={[Function]}
      onShareClick={[Function]}
    />
    <ForwardRef
      onClick={[Function]}
    />
  </ForwardRef>,
]
`;
